### PR TITLE
Switch to user defined CP module

### PR DESCRIPTION
**Pull request by jhgoh**
(edit: NHWC->NCHW done with dataset, NCHW=(batch,channel,eta,phi) checked by seunghwan)

- Switch to user-defined CP module.
- the input image (phi,eta) maps to (x,y): NCHW->batch,detector,eta,phi
- Default Dropout for the fully connected layers
- Use BCEWithLogitsLoss which is known to be numerically stable.

NOTE: Replication2d padding is not the CP, What we need is [1,2,3,4] -> [1,2,3,4,1,2...] but the RP gives [1,2,3,4] -> [1,...1,1,2,3,4,4...4]

### CODE REVIEW

**Comment by jhgoh** [comment]
Example: 
`>>> p = torch.nn.ReplicationPad2d((1,2))
>>> p(t)
tensor([[[0., 0., 1., 2., 2., 2.],
         [3., 3., 4., 5., 5., 5.],
         [6., 6., 7., 8., 8., 8.]]])
>>> t = torch.tensor([[[0.,1,2],[3,4,5],[6,7,8]]])
>>> p = torch.nn.ReplicationPad2d((1,2))
>>> t
tensor([[[0., 1., 2.],
         [3., 4., 5.],
         [6., 7., 8.]]])
>>> p(t)
tensor([[[0., 0., 1., 2., 2., 2.],
         [3., 3., 4., 5., 5., 5.],
         [6., 6., 7., 8., 8., 8.]]])
`

**Comment by jhgoh** [comment]
For the CP:
```
>>> cp = CircularPadY(2)
>>> cp(t)
tensor([[[0., 1., 2.],
         [3., 4., 5.],
         [6., 7., 8.],
         [0., 1., 2.],
         [3., 4., 5.],
         [6., 7., 8.]]])
```
